lib/common/client/configuration.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require 'singleton'
+
+module Common
+  module Client
+    class Configuration
+      include Singleton
```

Comment:
I like it as a separation of concern, and thought that a configuration is not something that should ever change for a client, so just load it into memory once and reuse it elsewhere. 

Eventually I would like to make a Rails Generator, that provided a set of questions, will build the skeleton of a new client for you, complete with specs. So just keeping things abstract to help facilitate that. I think we can revisit this though if the need should ever arise for configuration to be dynamic in nature.